### PR TITLE
Fix TypeError when exc_info=False

### PR DESCRIPTION
**Pull request by stj**
Formatting of a message fails in the `StdlibFormatter` when the
`exc_info` parameter is `False`. The traceback shows that handling of
the boolean is not done correct in the formatting for error type and
message.

```python
Traceback (most recent call last):
  File "lib/python3.8/logging/__init__.py", line 1081, in emit
    msg = self.format(record)
  File "lib/python3.8/logging/__init__.py", line 925, in format
    return fmt.format(record)
  File "ecs_logging/_stdlib.py", line 117, in format
    result = self.format_to_ecs(record)
  File "ecs_logging/_stdlib.py", line 163, in format_to_ecs
    value = extractors[field](record)
  File "ecs_logging/_stdlib.py", line 150, in <lambda>
    if (r.exc_info is not None and r.exc_info[0] is not None)
TypeError: 'bool' object is not subscriptable
```

This replaces the lambda with methods handling `exc_info` being an
boolean or iterable.

### CODE REVIEW

File: ecs_logging/_stdlib.py [hunk 1]
```diff
@@ -112,6 +112,24 @@ def __init__(self, stack_trace_limit=None, exclude_fields=()):
         self._exclude_fields = frozenset(exclude_fields)
         self._stack_trace_limit = stack_trace_limit
 
+    def _record_error_type(self, record):
```
Comment (sethmlarson) [inline ecs_logging/_stdlib.py:115]: Would be good to have an explict `return None` at the bottom of each of these functions

File: ecs_logging/_stdlib.py [hunk 2]
```diff
@@ -112,6 +112,24 @@ def __init__(self, stack_trace_limit=None, exclude_fields=()):
...
+        # upstream doesn't handle booleans for `exc_info`
+        if not record.exc_info:
```
Comment (sethmlarson) [inline ecs_logging/_stdlib.py:117]: Can we add comments for the two complex situations within `if` blocks below? Something about what the situation is when this branch is taken and how it's handled

File: ecs_logging/_stdlib.py [hunk 3]
```diff
@@ -112,6 +112,24 @@ def __init__(self, stack_trace_limit=None, exclude_fields=()):
...
+            return None
+        if isinstance(record.exc_info, bool):
+            return sys.exc_info()[0].__name__
```
Comment (sethmlarson) [inline ecs_logging/_stdlib.py:120]: Should we guard against `sys.exc_info()` returning `(None, None, None)`?

**Review by sethmlarson — CHANGES_REQUESTED** [review]
Here's some comments for you, going to tag in other reviewers.

File: ecs_logging/_stdlib.py (hunk 1 above)
Comment (stj) [reply ecs_logging/_stdlib.py:115]: done

File: ecs_logging/_stdlib.py (hunk 2 above)
Comment (stj) [reply ecs_logging/_stdlib.py]: Sure, should have removed that comment here too. It is in a subclass that I created to use `ecs_logging` in my project. It has no meaning here.

Pushed an update with more comments.

File: ecs_logging/_stdlib.py (hunk 3 above)
Comment (stj) [reply ecs_logging/_stdlib.py]: I refactored this a bit. Should be covered now.

**Comment by sethmlarson** [comment]
jenkins test this please

**Comment by sethmlarson** [comment]
jenkins test this please

**Review by basepi — APPROVED** [review]

**Pull request merged**